tests/publiccloud/sles4sap.pm

```diff
@@ -184,54 +184,81 @@ sub run {
                     die "DRBD device can't be found!" unless defined $drbd_device;
-                    my $drbd_uuid_device = $self->run_cmd(cmd => "blkid -o export $drbd_device | awk -F'=' '/^UUID=/ { print \$NF }'", quiet => 1);
-                    die "DRBD UUID can't be found!" unless defined $drbd_uuid_device;
-
-                    # Replace current device name by its UUID in config file and sync cluster conf
-                    $self->run_cmd(cmd => "sed -i '/[[:blank:]]*disk[[:blank:]]*\\/dev\\//s;\\(^[[:blank:]]*\\).*;\\1disk /dev/disk/by-uuid/${drbd_uuid_device}\\;;' $drbd_conf_file", quiet => 1);
+                    my $drbd_id_device = $self->run_cmd(cmd => "ls -l /dev/disk/by-id/ 2>/dev/null | awk '/\\/${drbd_device}\$/ { print \$9 }' | head -n1", quiet => 1);
+                    die "DRBD device ID can't be found!" unless defined $drbd_id_device;
+
+                    # All node should be configured
+                    my @nodes = split /\n/, $self->run_cmd(cmd => 'crm node show | sed -n \'/:[[:blank:]]*/s/(.*//p\'', quiet => 1);
+                    foreach my $node (@nodes) {
+                        # Replace current device name by its ID in config file and sync cluster conf
+                        $drbd_id_device =~ s/-0-part/-1-part/ unless $node eq $hostname;
+                        $self->run_cmd(cmd => "sed -i '/[[:blank:]]*disk[[:blank:]]*\\/dev\\//d' $drbd_conf_file", quiet => 1) if $node eq $hostname;
+                        $self->run_cmd(cmd => "sed -i '/on $node {/a\\disk /dev/disk/by-id/${drbd_id_device};' $drbd_conf_file", quiet => 1);
+                    }
+
+                    # Sync config and restart DRBD resource
                     $self->run_cmd(cmd => 'csync2 -xF', quiet => 1);
-
-                    # Cleanup DRBD resource
-                    $self->run_cmd(cmd => 'crm resource cleanup drbd-sapdata', quiet => 1);
+                    $self->run_cmd(cmd => 'crm resource restart drbd-sapdata', timeout => 120, quiet => 1);
                 }
 
                 # Workaround bsc#1179782 - Check if STONITH is disabled
-                unless ($instance->run_ssh_command(cmd => 'sudo sh -c \'crm configure show\' | grep -q stonith-enabled=false', rc_only => 1)) {
+                unless ($self->run_cmd(cmd => 'crm configure show | grep -q stonith-enabled=false', rc_only => 1)) {
                     record_soft_failure 'bsc#1179782 - [ha-sap-terraform-deployments_v6] stonith-enabled is set to false instead of true';
                     $self->run_cmd(cmd => 'crm configure property stonith-enabled=true', quiet => 1);
                 }
 
+                # Workaround bsc#1182701 on GCP - fence_gce doesn't restart the node as expected
+                if (is_gce) {
+                    # All node should be configured
+                    my @nodes = split /\n/, $self->run_cmd(cmd => 'crm node show | sed -n \'/:[[:blank:]]*/s/(.*//p\'', quiet => 1);
```

Comment:
This is the same as in L211? It can be defined before L200 as:
```
my @nodes = split /\n/, $self->run_cmd(cmd => 'crm node show | sed -n \'/:[[:blank:]]*/s/(.*//p\'', quiet => 1) if is_gce;
```
And you avoid sending the same command twice.

Reply:
Yes, but in the new code I'm using function for each workaround, so I kept as is.